### PR TITLE
attempt at fixing #3384

### PR DESCRIPTION
**Pull request by diml**
Split `camlp4.4.01.0` into `camlp4.4.01+system` for the system switch and `camlp4.4.01` for other switches

### CODE REVIEW

File: packages/camlp4/camlp4.4.01+system/opam [hunk 1]
```diff
@@ -0,0 +1,14 @@
+opam-version: "1.2"
+maintainer: "jeremie@dimino.org"
+homepage: "https://github.com/ocaml/camlp4"
+license: "LGPLv2"
+build: [
+  ["sh" "./check-camlp4.sh"]
+]
+available: [ preinstalled & (ocaml-version < "4.02") ]
```
Comment (samoht) [inline packages/camlp4/camlp4.4.01+system/opam]: shouldn't be ` ... & (ocaml-version >= "4.01")`

File: packages/camlp4/camlp4.4.01/opam [hunk 2]
```diff
@@ -3,7 +3,6 @@ maintainer: "jeremie@dimino.org"
 homepage: "https://github.com/ocaml/camlp4"
 license: "LGPLv2"
 build: [] # dummy package
-remove: []
-ocaml-version: [<"4.02.0"]
+available: [ (!preinstalled) & (ocaml-version < "4.02") ]
```
Comment (samoht) [inline packages/camlp4/camlp4.4.01/opam]: same here

File: packages/camlp4/camlp4.4.01+system/opam (hunk 1 above)
Comment (diml) [inline packages/camlp4/camlp4.4.01+system/opam]: Well, I assume we need this package for all versions of OCaml prior to 4.02. The version `4.01` is misleading though, would `camlp4.embed` or similar be OK for opam? I guess at worst we can use `camlp4.0`.

**Comment by samoht** [comment]
hum the solver seems confused:

```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed camlp4.4.01+system
Done.
[WARNING] External solver failed with empty solution Request saved to
          "/home/travis/.opam/log/solver-error-13536-1.cudf"
[ERROR] External solver failure, please fix your installation and check
        /home/travis/.opam/config and variable $OPAMEXTERNALSOLVER.
        You may also retry with option --use-internal-solver
```

@AltGr please help! :-)

**Comment by AltGr** [comment]
Ah... I removed this error in 1.2.2 I think, because it's annoying in some test cases (it was added because an empty solution could be dangerous -- remove everything -- but now that we control the solver-calling backend it's become unlikely to happen).
Workarounds could include --use-internal-solver, --no-base-packages, or installing some dummy packages. But in truth I don't get exactly what happens here. Why are there no base packages ?
And wtf is:

```
curl: option -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
```
???

**Comment by samoht** [comment]
Anyone has any idea of what's happening here? Seems that the scripts input get corrupted.

**Comment by samoht** [comment]
Can you try to rebase the PR to see if random reasons make it work better now?

**Comment by diml** [comment]
rebased

**Comment by diml** [comment]
Not sure what the last failure is, I can't see it on Travis...

**Comment by dsheets** [comment]
The only failure is the lint error for missing `authors` field which is called `default` right now.

**Comment by AltGr** [comment]
Yes, reports for @camelus are tagged "default": didn't find where to set the label. Code: https://github.com/AltGr/opam-ci/blob/96adac71238d50c74ce0150f02e14dd3b4a9f3e2/opam_ci.ml#L461-L463

Do you need to go through a registered github app rather than a direct token ?

**Comment by dsheets** [comment]
It's a missing field in the `new_status` type of the GitHub bindings. I've opened mirage/ocaml-github#88 to track.

**Pull request merged**